cmd/update-rke2-charts/main.go

```diff
@@ -0,0 +1,300 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/helm/pkg/chartutil"
+)
+
+var (
+	name    string
+	version string
+	gitSHA  string
+)
+
+const (
+	envCharts   = "CHARTS"
+	filePackage = "package.yaml"
+	fileChart   = "Chart.yaml"
+	fileValues  = "values.yaml"
+)
+
+const usage = `Version: %s
+%[2]s - rke2-charts version references update
+
+Usage: %[2]s [options] <chart name> [<version field>=<version value>]
+Options:
+  -h		show this help message
+  -c dir	rke2-charts directory (defaults: $CHARTS value or "rke2-charts")
+  -i		write changes into their respective files
+  -p		print resulting yaml files on STDOUT
+  -v		print version
+`
+
+func main() {
+	flag.Usage = func() {
+		w := os.Stderr
+		for _, arg := range os.Args {
+			if arg == "-h" {
+				w = os.Stdout
+				break
+			}
+		}
+		fmt.Fprintf(w, usage, version, name)
+	}
+
+	var (
+		vers, inPlace, printOut bool
+		chartsDir               string
+	)
+
+	chartDefault := "rke2-charts"
+	if c, ok := os.LookupEnv(envCharts); ok {
+		chartDefault = c
+	}
+
+	flag.BoolVar(&vers, "v", false, "print version")
+	flag.BoolVar(&inPlace, "i", false, "write changes into their respective files")
+	flag.BoolVar(&printOut, "p", false, "print resulting yaml files on STDOUT")
+	flag.StringVar(&chartsDir, "c", chartDefault, "rke2-charts directory")
+	flag.Parse()
+
+	if vers {
+		fmt.Fprintf(os.Stdout, "version: %s - git sha: %s\n", version, gitSHA)
+		return
+	}
+
+	args := flag.Args()
+
+	if len(args) == 0 {
+		exitErr("no chart provided")
+	}
+
+	chartName := flag.Arg(0)
+	chartPath := filepath.Join(chartsDir, "packages", chartName)
+
+	overrides := make(map[string]string, len(args))
```

Comment:
Actually, It should have been `len(args) -1` and I did it by mere force of habit. To my understanding the only benefit is to allocate the space you need right off the bat, as instead of letting the vm allocate an "small starting size" and re-allocate it later. 

That been said, on this context I don't think it might be impactful.

```
Map: An empty map is allocated with enough space to hold the
specified number of elements. The size may be omitted, in which case
a small starting size is allocated.
```
https://pkg.go.dev/builtin#make

Reply:
Generally, a "best practice".

Reply:
ok, thanks! I always create maps without length :)